File: PFinalQytetetJava/QytetetJava/src/modeloqytetet/Calle.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package modeloqytetet;

/**
 *
 * @author antonio-w10
 */
public class Calle extends Casilla {
    
    
    private TituloPropiedad titulo;

    public Calle(int numeroCasilla,TituloPropiedad titulo) {
        super(numeroCasilla, titulo);
        this.titulo=titulo;
    }
    
    
    TituloPropiedad asignarPropietario(Jugador jugador){
        this.titulo.setPropietario(jugador);
        return this.titulo;
    }
        
    
    protected TipoCasilla getTipo() {
        return TipoCasilla.CALLE;
    }

    protected TituloPropiedad getTitulo() {
        return titulo;
    }
    
    int pagarAlquiler(){
        int costeAlquiler= this.titulo.pagarAlquiler();
        return costeAlquiler;
    }
    
    protected boolean soyEdificable(){
        return true; //(getTipo() == TipoCasilla.CALLE)
    }
    
    
    private void setTitulo(TituloPropiedad titulo) {
        this.titulo = titulo;
    }
    
    
    
    boolean tengoPropietario(){
        return this.titulo.tengoPropietario();
    }
    
    
    @Override
    public String toString() {
        return super.toString() + 
                ", tipoCasilla=" + TipoCasilla.CALLE + ", titulo=" + titulo ;
    }
    
    
    
    
    
    
    
    
    
    
    
}

File: PFinalQytetetRuby/QytetetRuby/lib/calle.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module ModeloQytetet
  
  class Calle < Casilla
    
    attr_accessor :titulo
    
    def initialize(numeroCasilla, titulo)
      super(numeroCasilla,nil,nil, titulo)
      @titulo=titulo
    end
    
    def asignarPropietario(jugador)
      @titulo.propietario=(jugador)
      return @titulo
    end
    
    
    def pagarAlquiler()
      costeAlquiler = @titulo.pagarAlquiler()
      return costeAlquiler
    end
    
    def soyEdificable()
      return true
    end
        
    
    def tengoPropietario()
      return @titulo.tengoPropietario()
    end
    
    
    def to_s
      super + " \n titulo= #{@titulo}"
    end
       
    
    
    
  end
end
